### PR TITLE
[base] Fixed a data race in RootException.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

**Comment by mpimenov** [comment]
JTTP

File: base/exception.hpp [hunk 1]
```diff
@@ -8,24 +9,19 @@
...
-  {
-  }
+  RootException(char const * what, std::string const & msg);
 
   virtual ~RootException() throw()
   {
   }
 
-  virtual char const * what() const throw();
+  virtual char const * what() const throw() { return m_whatWithAscii.c_str(); }
```
Comment (VladiMihaylenko) [inline base/exception.hpp]: noexcept instead of throw()

Comment (VladiMihaylenko) [reply base/exception.hpp]: override

Comment (mpimenov) [reply base/exception.hpp]: done

**Pull request merged**